src/main/java/org/jclouds/vcloud/director/v1_5/compute/util/VCloudDirectorComputeUtils.java

```diff
@@ -182,32 +182,23 @@ public boolean apply(Network input) {
       });
    }
 
-
-   public static Optional<Network> tryFindNetworkNamed(final VCloudDirectorApi api, Org org, final String networkName) {
-      FluentIterable<Network> networks = FluentIterable.from(org.getLinks())
-              .filter(ReferencePredicates.typeEquals(ORG_NETWORK))
-              .transform(new Function<Link, Network>() {
+   public static Optional<Network> tryFindNetworkInVDCWithFenceMode(final VCloudDirectorApi api, Vdc vdc, final Network.FenceMode fenceMode) {
+      return FluentIterable.from(vdc.getAvailableNetworks())
+              .transform(new Function<Reference, Network>() {
                  @Override
-                 public Network apply(Link in) {
+                 public Network apply(Reference in) {
                     return api.getNetworkApi().get(in.getHref());
                  }
-              });
-
-      return tryFind(networks, new Predicate<Network>() {
+              })
+              .firstMatch(new Predicate<Network>() {
          @Override
          public boolean apply(Network input) {
             if (input.getTasks().size() != 0) return false;
```

Comment:
Looks a bit strange, but I think that's down to the way vcloud director models things. It's unclear why you lookup tasks, and whether doing that means you can now assume that getConfiguration().getFenceMode() will be non-null.

Perhaps marginally safer to do: `return fenceMode.equals(input.getConfiguration().getFenceMode())`